contracts/modules/erc4337/SafeProtocol4337Module.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+
+import {SafeProtocolAction, SafeTransaction} from "../../DataTypes.sol";
+import {PLUGIN_PERMISSION_EXECUTE_CALL} from "../../common/Constants.sol";
+import {IAccount} from "../../interfaces/Accounts.sol";
+import {ISafeProtocolManager} from "../../interfaces/Manager.sol";
+import {IERC165, ISafeProtocolFunctionHandler, ISafeProtocolPlugin} from "../../interfaces/Modules.sol";
+import {UserOperation} from "./interfaces/IERC4337.sol";
+import {ISafeProtocol4337Handler} from "./interfaces/ISafeProtocol4337Handler.sol";
+
+contract SafeProtocol4337Module is ISafeProtocolFunctionHandler, ISafeProtocolPlugin {
+    uint256 private constant VALIDATION_SIG_SUCCESS = 0;
+    uint256 private constant VALIDATION_SIG_FAILURE = 0;
+
+    address payable public immutable entrypoint;
+
+    constructor(address payable _entrypoint) {
+        require(_entrypoint != address(0), "invalid entrypoint address");
+        entrypoint = _entrypoint;
+    }
+
+    /**
+     * @inheritdoc IERC165
+     */
+    function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
+        return
+            interfaceId == type(IERC165).interfaceId ||
+            interfaceId == type(ISafeProtocolFunctionHandler).interfaceId ||
+            interfaceId == type(ISafeProtocolPlugin).interfaceId;
+    }
+
+    /**
+     * @inheritdoc ISafeProtocolFunctionHandler
+     */
+    function handle(address account, address sender, uint256 value, bytes calldata data) external override returns (bytes memory result) {
+        require(sender == entrypoint, "unsupported entrypoint");
+        require(value == 0, "not payable");
```

Comment:
This value variable is unused; does it make sense to omit the check and remove the argument?

Reply:
This is part of the interface of the `ISafeProtocolFunctionHandler`. I think we could change it to not have a value, but that would be strange to me:
1. I think fallbackhandlers should be able to emulate non-`payable` functions (so revert on `value != 0`
2. `payable` fallback handlers should be able to read the amount of `Ether` that was sent with the transaction to the account.

I would also argue this is a bit out of the scope of this PR.